provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/http/ContractAPIHttp.java

```diff
@@ -211,7 +217,8 @@ public synchronized void requestNonceForTPM(RequestData requestData, ResponseCal
                 //SRS_ContractAPIHttp_25_008: [If service return a status as 404 then this method shall trigger the callback to the user with the response message.]
                 if (httpResponse.getStatus() == ACCEPTABLE_NONCE_HTTP_STATUS)
                 {
-                    TpmRegistrationResultParser registerResponseTPMParser = TpmRegistrationResultParser.createFromJson(new String(e.getMessage()));
+                    String tpmRegistrationResultJson = new String(httpResponse.getErrorReason(), StandardCharsets.UTF_8);
```

Comment:
The exception message that we used to throw and that we gave to this parser object was just the http response error reason payload. This parser still gets that same error reason, but now our thrown exception has the status code as well. If I left this line as e.GetMessage(), the tpm parser would throw because it doesn't expect a status code